hwbench/helpers/display.py

```diff
@@ -0,0 +1,33 @@
+from typing import Optional
+
+
+def cpu_list_to_range(cpu_list: list[int]) -> str:
+    """
+    This function takes a list of integers as input and will link them together in a nicely formatted string
+    - `[0, 1, 2, 3, 4, 5]` will give `"0-5"`
+    - `[0, 4, 2, 3, 7, 8, 9]` will give `"0, 2-4, 7-9"`
+
+    It was made specifically for formatting a CPU cores list
+    """
+    cpu_list.sort()
+    output: list[str] = []
+    previous_entry: Optional[int] = cpu_list[0]
+
+    for i in range(1, len(cpu_list)):
+        current_entry = cpu_list[i]
+        is_immediately_next = current_entry - 1 == cpu_list[i - 1]
+        needs_compression = cpu_list[i - 1] != previous_entry
+
+        if not is_immediately_next:
+            if needs_compression:
+                output.append(f"{previous_entry}-{cpu_list[i-1]}")
+            else:
+                output.append(str(previous_entry))
+            previous_entry = current_entry
+        if i == len(cpu_list) - 1:
```

Comment:
Please add a comment to indicate this part is to manage the last digit on the cpu_list